### DISCUSSION TITLE
Rotating RectGrid around y-axis

### DISCUSSION THREAD

**Discussion by MarieH855** [acoular/acoular#401] [Q&A] [answered]
Hi

I'm currently working on localising sound sources on wind turbines while using acoular. I gave a presentation on this topic on Thursday at the Acoustic Seminar at TU Berlin.
	
As the wind direction was not optimal in the measurements, the plane in which the rotor blades move is not parallel to the x-y plane of the coordinate system used for the evaluation, but rotated by a few degrees around the y-axis. My current approaches for a fixed grid are either to create a customised grid using ImportGrid or to transform the entire coordinate system, in order to have the z-axis perpenicular to the measurement plane (the origin of the coordinate system would no longer be located within the microphone array) and then using RectGrid. Is there a way to perform the rotation directly on the RectGrid object?
	
The next step would be to perform beamforming using a moving focus. That would probably be easiest with a RectGrid object or is this also possible when using an ImportGrid object?

**Comment by esarradj** [comment, marked as answer]
For a fixed grid I would use `RectGrid` to compute the grid positions and then write the rotated position into an empty `ImportGrid`.

For a moving grid this not necessary, you can just set the axis perpendicular to the grid by setting `BeamformerTimeTraj.rvec`.
